SDL/Event.cu

```diff
@@ -722,228 +754,226 @@ void SDL::Event::addHitToEvent(std::vector<float> x, std::vector<float> y, std::
+    cudaMemcpyAsync(segmentsInGPU->etaErr,&etaErr[0],size*sizeof(float),cudaMemcpyHostToDevice,stream);
+    cudaMemcpyAsync(segmentsInGPU->eta,&eta[0], size*sizeof(float),cudaMemcpyHostToDevice,stream);
+    cudaMemcpyAsync(segmentsInGPU->phi,&phi[0], size*sizeof(float),cudaMemcpyHostToDevice,stream);
+    cudaMemcpyAsync(segmentsInGPU->charge,&charge[0], size*sizeof(int),cudaMemcpyHostToDevice,stream);
+    cudaMemcpyAsync(segmentsInGPU->superbin,&superbin[0],size*sizeof(int),cudaMemcpyHostToDevice,stream);
+    cudaMemcpyAsync(segmentsInGPU->pixelType,&pixelType[0],size*sizeof(int8_t),cudaMemcpyHostToDevice,stream);
+    cudaMemcpyAsync(segmentsInGPU->isQuad,&isQuad[0],size*sizeof(short),cudaMemcpyHostToDevice,stream);
     cudaStreamSynchronize(stream);
-    unsigned int nThreads = 256;
-    unsigned int nBlocks =  MAX_BLOCKS;//size % nThreads == 0 ? size/nThreads : size/nThreads + 1;
-
-    addPixelSegmentToEventKernel<<<nBlocks,nThreads,0,stream>>>(hitIndices0_dev,hitIndices1_dev,hitIndices2_dev,hitIndices3_dev,dPhiChange_dev,ptIn_dev,ptErr_dev,px_dev,py_dev,pz_dev,eta_dev, etaErr_dev, phi_dev, charge_dev, pixelModuleIndex, *modulesInGPU, *rangesInGPU, *hitsInGPU,*mdsInGPU,*segmentsInGPU,size, superbin_dev, pixelType_dev,isQuad_dev);
-
-   //cudaDeviceSynchronize();
-   cudaStreamSynchronize(stream);
-   cudaMemcpyAsync(&(segmentsInGPU->nSegments)[pixelModuleIndex], &size, sizeof(unsigned int), cudaMemcpyHostToDevice,stream);
-   cudaMemcpyAsync(&(segmentsInGPU->totOccupancySegments)[pixelModuleIndex], &size, sizeof(unsigned int), cudaMemcpyHostToDevice,stream);
-   unsigned int mdSize = 2 * size;
-   cudaMemcpyAsync(&(mdsInGPU->nMDs)[pixelModuleIndex], &mdSize, sizeof(unsigned int), cudaMemcpyHostToDevice,stream);
-   cudaMemcpyAsync(&(mdsInGPU->totOccupancyMDs)[pixelModuleIndex], &mdSize, sizeof(unsigned int), cudaMemcpyHostToDevice,stream);
-   cudaStreamSynchronize(stream);
-  
+
+    // Temporary fix for queue initialization.
+    QueueAcc queue(devAcc);
+
+    Idx const size_elements(size);
+    alpaka::Vec<Dim, Idx> const pixel_segments_extent(size_elements);
+
+    alpaka::WorkDivMembers<Dim, Idx> const pixel_segments_workdiv(
+        alpaka::getValidWorkDiv<Acc>(
+            devAcc,
+            pixel_segments_extent,
+            elementsPerThread,
+            false,
+            alpaka::GridBlockExtentSubDivRestrictions::Unrestricted));
+
+    addPixelSegmentToEventKernel pixel_segments_kernel;
+    auto const pixel_segments_task(alpaka::createTaskKernel<Acc>(
+        pixel_segments_workdiv,
+        pixel_segments_kernel,
+        hitIndices0_dev,
+        hitIndices1_dev,
+        hitIndices2_dev,
+        hitIndices3_dev,
+        dPhiChange_dev,
+        pixelModuleIndex,
+        *modulesInGPU,
+        *rangesInGPU,
+        *hitsInGPU,
+        *mdsInGPU,
+        *segmentsInGPU,
+        size_elements));
+
+    alpaka::enqueue(queue, pixel_segments_task);
+    alpaka::wait(queue);
```

Comment:
Out of curiosity, does this replace a `cudaStreamSynchronize(stream);`?

- If yes, then that's fine, it closely follows the previous implementation. However, it's a bit weird that later (L970) we also use `cudaStreamSynchronize(stream);`. Is there some detail here (synchronize stream vs. wait for queue) that we need to pay more attention to?
- If no, is there a reason we dropped the previous `cudaStreamSynchronize(stream);` from this implementation?

Reply:
Yes, wait is equivalent to stream synchronize. This PR only replaces the kernel code though. All other code still uses CUDA streams, which is why the Synchronizes are necessary.